airbyte-ci/connectors/metadata_service/lib/tests/fixtures/__init__.py

```diff
@@ -7,28 +7,41 @@
 
 def list_all_paths_in_fixture_directory(folder_name: str) -> List[str]:
     file_path = os.path.join(os.path.dirname(__file__), folder_name)
+
+    # If folder_name has subdirectories, os.walk will return a list of tuples,
+    # one for folder_name and one for each of its subdirectories.
+    fixture_files = []
     for root, dirs, files in os.walk(file_path):
-        return [os.path.join(root, file_name) for file_name in files]
+        fixture_files.extend(os.path.join(root, file_name) for file_name in files)
+    return fixture_files
 
 
 @pytest.fixture(scope="session")
 def valid_metadata_yaml_files() -> List[str]:
-    return list_all_paths_in_fixture_directory("metadata_validate/valid")
+    files = list_all_paths_in_fixture_directory("metadata_validate/valid")
+    assert len(files) > 0, "No files found in metadata_validate/valid"
```

Comment:
```suggestion
   if not files:
      pytest.fail("No files found in metadata_validate/valid")
```
I prefer to use this more explicit way of failing a tests when its fixture can't be generated.

Reply:
Oh nice, didn't think about `pytest.fail()`. On it!